src/main/java/com/jcabi/beanstalk/maven/plugin/AbstractBeanstalkMojo.java

```diff
@@ -259,6 +332,82 @@ protected boolean isGreen(final Environment env) {
     }
 
     /**
+     * Validate given text in YAML format.
+     *
+     * @param text YAML text
+     */
+    protected void validYaml(final String text) {
+        new Yaml().load(text);
+    }
+
+    /**
+     * Validate the war file.
+     *
+     * @param zip Zip file
+     * @throws org.apache.maven.plugin.MojoFailureException Thrown, if
+     *  validation fails.
+     */
+    protected void validate(final ZipFile zip) throws MojoFailureException {
+        if (zip.getEntry(".ebextension") == null) {
+            throw new MojoFailureException(
+                ".ebextensions directory does not exist in the WAR file"
+            );
+        }
+        final Enumeration<? extends ZipEntry> entries = zip.entries();
+        while (entries.hasMoreElements()) {
+            final ZipEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(".ebextensions/")
+                && !entry.isDirectory()) {
+                final String text = this.readFile(zip, entry);
+                try {
+                    this.validYaml(text);
+                } catch (final YAMLException exception) {
+                    final String msg =
+                        "File '%s' in .ebextensions is not YAML valid. %s";
+                    throw new MojoFailureException(
+                        String.format(
+                            msg,
+                            entry.getName(),
+                            exception.getMessage()
+                        ),
+                        exception
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads text from a ZIP file.
+     * @param warf ZIP file, which contains entry.
+     * @param entry ZIP entry (compressed file) to read from.
+     * @return Text content of entry.
+     * @throws MojoFailureException thrown when encounter error.
+     */
+    private String readFile(final ZipFile warf, final ZipEntry entry)
+        throws MojoFailureException {
+        InputStream stream = null;
+        InputStreamReader reader = null;
```

Comment:
@tmjee this variable `stream` is not required. When you close a reader, stream is being closed automatically. You can just create an instance of `InputStreamReader` and then close it in `finally`. The stream will be closed too.